kernel/common/kernel.rb

```diff
@@ -504,7 +504,7 @@ def load(name, wrap=false)
   module_function :load
 
   def loop
-    return to_enum(:loop) unless block_given?
+    return to_enum(:loop) { Float::INFINITY } unless block_given?
```

Comment:
Interesting difference: `to_enum(:loop).size` returns `nil` on MRI 2.2, but using `Kernel.loop.size` returns `Infinity`. As far as I can tell this is implementing `Kernel.loop` so this should be fine.